### PR TITLE
[acr] login: support cross tenant login

### LINKED ISSUE #7649: `az acr login` does not work with `--subscription` argument
**Describe the bug**
`az acr login` fails if value of `--subscription` argument does not match active subscription as shown by `az account show`.

**To Reproduce**

Set active subscrption 

```
az account set --subscription SOME_SUB_NAME
```

Try `acr login` into another sub.
Observe "Access denied" message.
I then get prompted for username.

```
az acr login --name REG_NAME --resource-group utv --subscription REQUIRED_SUB_NAME
Unable to get AAD authorization tokens with message: Access to registry 'REG_NAME.azurecr.io' was denied. Response code: 401.
Username:
```

Set active sub to be the same as required target sub.
Observe everything works now.

```
az account set --subscription REQUIRED_SUB_NAME
az acr login --name REG_NAME --subscription REQUIRED_SUB_NAME
Login Succeeded
```

**Environment summary**
```
az --version
azure-cli (2.0.48)
```

### PR DESCRIPTION
**Pull request by adewaleo**
closes #7649

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acr/import.py [hunk 1]
```diff
@@ -90,3 +94,12 @@ def acr_import(cmd,
         resource_group_name=resource_group_name,
         registry_name=registry_name,
         parameters=import_parameters)
+
+
+def _validate_target_tags(target_tags):
+    if not target_tags:
+        return
+
+    for t_tag in target_tags:
```
Comment (adewaleo) [inline src/azure-cli/azure/cli/command_modules/acr/import.py]: Simple validation to deal with the user-input issue @SteveLasker found. cc: @sajayantony @djyou 

Does this look good or is this overkill @sajayantony 

target tags warning

File: src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py [hunk 2]
```diff
@@ -238,6 +242,7 @@ def _get_credentials(cmd,  # pylint: disable=too-many-statements
 
     # 2. if we don't yet have credentials, attempt to get a refresh token
     if not registry or registry.sku.name in get_managed_sku(cmd):
+        logger.info("Attempting to retrieve AAD refresh token...")
```
[the remark this answers is not in the document]
Comment (sajayantony) [reply src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py]: Wouldn't we move this upto the RP. Say we implement this in the portal would we have to do the same kind of validation?

File: src/azure-cli/azure/cli/command_modules/acr/import.py (hunk 1 above)
Comment (djyou) [reply src/azure-cli/azure/cli/command_modules/acr/import.py]: I would recommend having another PR for the import issue and let this one focus on cross tenant login.

Comment (adewaleo) [reply src/azure-cli/azure/cli/command_modules/acr/import.py]: Done!

File: src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adewaleo) [reply src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py]: > Wouldn't we move this upto the RP. Say we implement this in the portal would we have to do the same kind of validation?

@sajayantony Do you mean the logging here or the steps to retrieve credentials? Sorry just seeing this.

**Pull request merged**